cmd/format-erasure.go

```diff
@@ -545,43 +543,32 @@ func formatErasureFixLocalDeploymentID(endpoints Endpoints, storageDisks []Stora
 
 // Get backend Erasure format in quorum `format.json`.
 func getFormatErasureInQuorum(formats []*formatErasureV3) (*formatErasureV3, error) {
-	formatHashes := make([]string, len(formats))
-	for i, format := range formats {
-		if format == nil {
-			continue
-		}
-		h := sha256.New()
-		for _, set := range format.Erasure.Sets {
-			for _, diskID := range set {
-				h.Write([]byte(diskID))
-			}
-		}
-		formatHashes[i] = hex.EncodeToString(h.Sum(nil))
-	}
-
 	formatCountMap := make(map[string]int)
-	for _, hash := range formatHashes {
-		if hash == "" {
+	for _, format := range formats {
+		if format == nil {
 			continue
 		}
-		formatCountMap[hash]++
+		formatCountMap[format.ID]++
```

Comment:
I think in practice format.ID will always be the same in all disks - in updated or outdated disks - this doesn't seem enough to get a quorum format

Reply:
It is sufficient because deployment Id is always same for same sets. 

It is impossible to have same deployment I'd and different sets.

The implementation that we have today is unnecessarily expensive.

Reply:
I am not sure I understand this - 

getFormatErasureInQuorum() is supposed to return format.json which has the most quorum in all disks - AFAIK a disk can be outdated - let's say we re-formatted disks when one node is offline - we should not rely on that outdated information anymore. (let's discuss offline)